### ISSUE TITLE
Article search does not work on Kiwix 2.4 Build 1182160, Android 5.1.1

### ISSUE THREAD

**Issue by uwemock** [kiwix/kiwix-android#833]
Device: Samsung Galaxy Tab A6, Android 5.1.1
Kiwix 2.4 Build 1182160
ZIM: wikipedia_de_all_nopic_2018-05 in 2 GB chunks

Steps to reproduce:
1. Open Kiwix
2. Tap overflow menu and select "Get Content"
3. Wait until file search is complete
4. Select "Wikipedia" from files list
5. Wait until Wikipedia start page appears
6. Tap looking glass icon in action bar
7. From the keyboard, tap "M"

Expected behaviour: List of "M*" headlines appears: "M", "M & A", "M & M",...
Observed behaviour: "m/fe14645d633efb57d07e760170c44f83a8769ee1.svg" appears

8. From the keyboard, tap "E"

Expected behaviour: List of "ME*" headlines appears: "Me", "Me & Mrs. Jones", ...
Observed behaviour: Nothing appears

On my Samsung S5 mini, Android 6.0.1, these steps only give the expected result when the app data has been deleted prior to starting the app. If the app has run before, random things may happen: Kiwix starting up properly showing the most recently used file. OR Kiwix starting up with the welcome screen, upon selecting the most recently used ZIM file the app will complain that it is is now ""not a vaild ZIM file". OR Kiwix starting up with the welcome screen, upon selecting the most recently used ZIM file the app will open it with no problem.

**Comment by mhutti1** [comment]
Download platform tools from https://developer.android.com/studio/releases/platform-tools

Connect phone to computer. Run command adb logcat
Copy result/pipe to file and post.

**Comment by mhutti1** [comment]
I think the issue is:
```
W/kiwix   ( 5121): Unable to get url for title m/fe14645d633efb57d07e760170c44f83a8769ee1.svg: 
W/kiwix   ( 5121): std::exception
 ```

That looks like a very strange URL to me. @kelson42 any ideas?

**Comment by kelson42** [comment]
@mhutti1 Yes, quite strange why the WE search a title for this image url. At this level the behaviour might be correct, the question is more how this string `m/fe14645d633efb57d07e760170c44f83a8769ee1.svg` cam here in first place.

**Comment by uwemock** [comment]
I expected things to work properly on my phone. I found yesterday that this is not the case.

I searched for "DFB-Pokal" in the german Wikipedia. When I enter the search expression, there are only 10 results for "DF" and no result for "DFB". In the online Wikipedia, there are more than 10 articles that contain "DFB" in their title.

**Comment by SaxSeven** [comment]
Same to me. I installed Kiwix 2.4 build 1182160 ( on my Samsung S9 )wikipedia_de_all_nopic_2018-05.zim,splitted it into 2GB chunks.
The start page appears, all links work, but the Search doesnt work at all.
Best regards Gunter

**Comment by TomTomH2** [comment]
Same problem here!
ZIM: wikipedia_de_all_nopic_2018-05 in 2 GB chunks
latest kiwix 2.4 (182160)

Samsung Galaxy Note 4 on Android 6.0.1
wikipedia_de_all_nopic_2018-05, split into 2GB-chunks: Search doesn't work at all, only intra-page
wikipedia_en_medicine_novid_2018-08: Search works normally

**Comment by SaxSeven** [comment]
is anybody from kiwix taking care ??

**Comment by kelson42** [comment]
@SaxSeven We would be happy to fix it... the problem is that we have so far no reproduction case!

**Comment by Imarok** [comment]
I have the same problem as the OP.
No search for wikipedia_de_all_nopic_2018-05. Searching in e.g. wikiquote works.
Just to be sure: @kelson42 have you tried reproducing it exactly with wikipedia_de_all_nopic_2018-05?

**Comment by kelson42** [comment]
What would be helpful at this stage is to explain how you have splited your files and if possible post the list of files with the md5sum or similar.

**Comment by Imarok** [comment]
Just to make sure: I have not split the zim file into 2GB chunks. (At least not intentionally)
I just downloaded it from [here](http://wiki.kiwix.org/wiki/Special:MyLanguage/Content_in_all_languages) via torrent and then copied it onto my phone.

**Comment by Popolechien** [comment]
Adding another log from another user, FYI
[logcat.txt](https://github.com/kiwix/kiwix-android/files/2547487/logcat.txt)

@Imarok Yes it does, at least for the larger wikis. We also have the occasional server crash, which is annoying when that happens after 2+ weeks of compressing 0_o

**Comment by kelson42** [comment]
This should have been fixed in Kiwix Android 3.0. Please reopen if not.
